src/allotropy/parsers/moldev_softmax_pro/softmax_pro_structure.py

```diff
@@ -445,55 +369,53 @@ def create(
             reduced_data=TimeReducedData.create(reader),
         )
 
-    def get_raw_data(self) -> TimeRawData:
-        return assert_not_none(
+    def iter_wavelengths(self, position: str) -> Iterator[DataElement]:
+        raw_data = assert_not_none(
             self.raw_data,
             msg="Unable to find plate block raw data.",
         )
 
-    def get_reduced_data(self) -> TimeReducedData:
-        return assert_not_none(
-            self.reduced_data,
-            msg="Unable to find plate block reduced data.",
-        )
+        for wavelength_data in raw_data.wavelength_data:
+            for kinetic_data in wavelength_data.kinetic_data:
+                yield DataElement(
+                    temperature=kinetic_data.temperature,
+                    wavelength=wavelength_data.wavelength,
+                    position=position,
+                    value=kinetic_data.data[position],
+                )
 
 
 @dataclass(frozen=True)
 class PlateBlock(Block):
     block_type: str
     header: PlateHeader
-    well_data: defaultdict[str, WellData]
+    block_data: Union[PlateData, TimeData]
+    plate_block_type: str = "Abstract"
 
     @staticmethod
     def create(reader: CsvReader) -> PlateBlock:
         header_series = PlateBlock.read_header(reader)
         cls = PlateBlock.get_plate_block_cls(header_series)
         header = cls.parse_header(header_series)
-        well_data: defaultdict[str, WellData] = defaultdict(WellData.create)
 
         if header.export_format == ExportFormat.TIME_FORMAT.value:
-            PlateBlock._parse_time_format_data(
-                header,
-                well_data,
-                time_data=TimeData.create(reader, header),
+            return cls(
+                block_type="Plate",
+                raw_lines=reader.lines,
+                header=header,
+                block_data=TimeData.create(reader, header),
             )
         elif header.export_format == ExportFormat.PLATE_FORMAT.value:
-            PlateBlock._parse_plate_format_data(
-                header,
-                well_data,
-                plate_data=PlateData.create(reader, header),
+            return cls(
```

Comment:
out of scope for fixing in this PR, but having a static method in a parent class to potentially instantiate children is pretty ugly. I think this is a known issue though.